src/components/Dialogs/Dialogs.jsx

```diff
@@ -1,39 +1,57 @@
 import React from "react";
+import { NavLink } from "react-router-dom";
+
+import { ROUTES, I18N } from "../core/constants";
 
 import s from "./Dialogs.module.css";
 
-const Dialogs = (props) => {
+const DialogItem = props => {
+  let path = ROUTES.MESSAGES + "/" + props.id;
+
+  return (
+    <div className={s.dialog + " " + s.active}>
+      <NavLink to={path}>{props.name}</NavLink>
+    </div>
+  );
+};
+
+const Message = props => {
+  return <div className={s.dialogs}>{props.message}</div>;
+};
+
+const Dialogs = props => {
```

Comment:
Each time the Dialogs component is rendered or re-rendered, this functional component is called.
Therefore all of the variables inside of it are re-created - it is a normal behaviour for all the functions in JS.
In the case of Dialogs component it means that the "dialogsData" and "messageData" are created over and over again - but they don't change each time! That means we could store them OUTSIDE the functional component - as you did in ProfileInfo component with PROFILE_LINK.

Why is it useful, you ask?
Imagine these "dialogsData" and "messageData" contain 1000+ of objects of data inside - that will decrease the performance of the Dialogs component.

Reply:
all consts moved to core folder